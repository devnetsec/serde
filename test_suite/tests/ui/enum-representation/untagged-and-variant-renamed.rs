use serde_derive::{Serialize, Deserialize};

#[derive(Serialize, Deserialize)]
#[serde(untagged)]
enum E {
	#[serde(alias = "different-name")]
	A(u8),
	#[serde(rename = "different-name")]
	B(String),
}
fn main() {}
