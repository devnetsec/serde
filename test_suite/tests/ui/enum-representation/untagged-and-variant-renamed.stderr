error: adding a alias to a variant of an untagged enum does nothing
 --> tests/ui/enum-representation/untagged-and-variant-renamed.rs:6:2
  |
6 |     #[serde(alias = "different-name")]
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: renaming a variant of an untagged enum does nothing
 --> tests/ui/enum-representation/untagged-and-variant-renamed.rs:8:2
  |
8 |     #[serde(rename = "different-name")]
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
